utils/data.ts

```diff
@@ -43,6 +43,8 @@ export const isEqual = (a: unknown, b: unknown): boolean => {
   }
   if (isSymbol(a) && isSymbol(b)) return a.toString() === b.toString();
   if (isObject(a) && isObject(b)) return JSON.stringify(a) === JSON.stringify(b);
+  if (Array.isArray(a) && Array.isArray(b))
+    return a.length === b.length && a.every((x, i) => isEqual(x, b[i]));
   return a === b;
```

Comment:
I wonder whether this is faster or treating array as objects and stringifying them. what caveats would that have?

Reply:
im assuming this would be faster since stringifying is one pass then string comparison is another?

also this way if the first element is not equals it would go ahead and break out

guess it depends on how `isEquals` handles the individual elements though. if it ends up stringifying each individual element separate, i can imagine that might be slower in cases where things are equal.

Reply:
the recursive calls we have to make, give me pause. because of how `.every` behaves and the early breaking it probably still is a net benefit 🤔

Reply:
ah I just reloaded didn't see your comment before

Reply:
i think in general if its performance sensitive, it's better to use a custom `isEquals`  that takes into account your knowledge of the underlying data

Reply:
i mainly just want `isEquals([], [])` to be `true`, don't really care that much otherwise.

Reply:
on the topic: `JSON.stringify` can throw an error if there's cyclical references. don't think it's come up, but might be worth handling at some point.

Reply:
> i mainly just want isEquals([], []) to be true, don't really care that much otherwise.

I imagine this case is hit very often as well. good catch.